packages/manager/apps/pci-load-balancer/src/components/detail/listeners/ListenerForm.page.tsx

```diff
@@ -0,0 +1,320 @@
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_INPUT_TYPE,
+  ODS_SPINNER_SIZE,
+  ODS_TEXT_LEVEL,
+} from '@ovhcloud/ods-components';
+import {
+  OsdsButton,
+  OsdsFormField,
+  OsdsInput,
+  OsdsSelect,
+  OsdsSelectOption,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  LISTENER_POOL_PROTOCOL_COMBINATION,
+  PROTOCOLS,
+  PROTOCOLS_PORT,
+} from '@/constants';
+import {
+  TLoadBalancerListener,
+  TLoadBalancerPool,
+} from '@/api/data/load-balancer';
+import Popover from './Popover';
+
+type TFormState = {
+  name: string;
+  protocol: string;
+  port: number;
+  pool: TLoadBalancerPool | null;
+};
+
+export type ListenerFormProps = {
+  listener?: TLoadBalancerListener | null;
+  pools: TLoadBalancerPool[];
+  isPending: boolean;
+  isEditing?: boolean;
+  onSubmit: (state: TFormState) => void;
+  onCancel: () => void;
+};
+
+export default function ListenerForm({
+  listener = null,
+  isEditing = false,
+  pools,
+  isPending,
+  onCancel,
+  onSubmit,
+}: ListenerFormProps) {
+  const { t: tCommon } = useTranslation('pci-common');
+  const { t } = useTranslation('octavia-load-balancer-listeners-create');
+  const { t: tListeners } = useTranslation('octavia-load-balancer-listeners');
+
+  const PORT_MIN_VALUE = 1;
+  const PORT_MAX_VALUE = 65535;
+
+  const [formState, setFormState] = useState<TFormState>({
+    name: '',
+    protocol: '',
+    port: 1,
+    pool: null,
+  });
+
+  const [filteredPools, setFilteredPools] = useState<TLoadBalancerPool[]>([]);
+  const [isTouched, setIsTouched] = useState({ name: false, port: false });
+
+  const getFilteredPools = (): TLoadBalancerPool[] =>
+    pools.filter(
+      ({ listenerId, protocol }) =>
+        !listenerId &&
+        LISTENER_POOL_PROTOCOL_COMBINATION[formState.protocol]?.includes(
+          protocol,
+        ),
+    ) || [];
+
+  useEffect(() => {
+    if (isEditing && listener && pools?.length) {
+      setFormState({
+        name: listener.name,
+        protocol: listener.protocol,
+        port: listener.port,
+        pool: pools.find((pool) => pool.id === listener.defaultPoolId) ?? null,
+      });
+    }
+  }, [listener, isEditing, pools]);
+
+  useEffect(() => {
+    if (pools) {
+      setFilteredPools(formState.protocol ? getFilteredPools() : pools);
+    }
+  }, [pools, formState.protocol]);
+
+  const handleProtocolChange = (event) => {
```

Comment:
handle functions could be grouped into a handle object:
`const handle = {
  protocolChange : ()=>{
    //implementation
  },
  change : ()=>{
    //implementation
  },
  blur : ()=>{
    //implementation
  }
}`

for more lisibility

Reply:
Done